File: Data Structures & Algorithms/find-duplicate-integer/submission-3.js
class Solution {
    /**
     * @param {number[]} nums
     * @return {number}
     */
    findDuplicate(nums) {
        console.log(nums)
        for (let i = 0; i < nums.length; i++) {
            const index = Math.abs(nums[i]) - 1;
            console.log(nums[index], nums[i])
            if (nums[index] < 0)
                return Math.abs(nums[i])
            else
            {
                nums[index] *= -1;
            }
        }
        console.log(nums)
    }
}

File: Data Structures & Algorithms/find-k-closest-elements/submission-3.py
class Solution:
    def findClosestElements(self, arr: List[int], k: int, x: int) -> List[int]:
        if x <= arr[0]:
            return arr[:k]
        if x >= arr[-1]:
            return arr[len(arr) - k:]
        l = 0
        r = len(arr) - 1
        while r - l + 1 > k:
            if abs(arr[l] - x) > abs(arr[r] - x):
                l += 1
            else:
                r -= 1
        return arr[l:r + 1]
                


            

            


